### PR TITLE
meson: clean up lastlog2 handling a bit

### PR DESCRIPTION
**Pull request by t-8ch**
See #2754 

@eworm-de Is this sufficient?

### CODE REVIEW

**Comment by eworm-de** [comment]
No, still fails, even if I give `-Dbuild-liblastlog2=disabled -Dbuild-pam-lastlog2=disabled`...

```
[381/611] Linking target liblastlog2/liblastlog2.so.1.3.0
FAILED: liblastlog2/liblastlog2.so.1.3.0 
gcc -m32  -o liblastlog2/liblastlog2.so.1.3.0 liblastlog2/liblastlog2.so.1.3.0.p/src_lastlog2.c.o -flto -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,--start-group -Wl,-soname,liblastlog2.so.1 -Wl,--end-group -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -flto=auto -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/lib32-util-linux/src=/usr/src/debug/lib32-util-linux -flto=auto -Wl,--version-script=/build/lib32-util-linux/src/util-linux/liblastlog2/src/liblastlog2.sym
/usr/bin/ld: /tmp/cc8XIsk8.ltrans0.ltrans.o: in function `open_database_ro':
/usr/src/debug/lib32-util-linux/build/<artificial>:82:(.text+0x37): undefined reference to `sqlite3_open_v2'
/usr/bin/ld: /usr/src/debug/lib32-util-linux/build/<artificial>:85:(.text+0x55): undefined reference to `sqlite3_errmsg'
/usr/bin/ld: /tmp/cc8XIsk8.ltrans0.ltrans.o: in function `open_database_ro':
/usr/src/debug/lib32-util-linux/build/../util-linux/liblastlog2/src/lastlog2.c:89:(.text+0x84): undefined reference to `sqlite3_close'
[...]
```

**Comment by t-8ch** [comment]
Please try again

**Comment by eworm-de** [comment]
With these four commit applied the build succeeds. Thanks a lot!

File: liblastlog2/meson.build [hunk 1]
```diff
@@ -50,9 +48,9 @@ if build_liblastlog2
     exe = executable(
       test_name,
       'src/tests/tst_' + lastlog2_test + '.c',
-      include_directories : [dir_include, dir_liblastlog2],
-      link_with : [lib_common, lib_lastlog2],
-      dependencies : libdl
+      include_directories : [dir_include],
```
Comment (karelzak) [inline liblastlog2/meson.build:51]: Not sure if this is correct. We need to install the header file into "liblastlog2" subdirectory. For other libs, we use, for example:
```
include_directories : [dir_include, dir_libblkid]
```
The usual final path should be _/usr/inlude/liblastlog2/lastlog2.h_ and in lastlog2.pc need to use `Cflags: -I${includedir}/liblastlog2`

Comment (t-8ch) [reply liblastlog2/meson.build:51]: The new dependency below carries both the library and the same include dir.

**Pull request merged**